go/vt/discovery/topology_watcher_test.go

```diff
@@ -12,12 +13,26 @@ import (
 )
 
 func TestCellTabletsWatcher(t *testing.T) {
-	ft := newFakeTopo()
+	checkWatcher(t, true)
+}
+
+func TestShardReplicationWatcher(t *testing.T) {
+	checkWatcher(t, false)
+}
+
+func checkWatcher(t *testing.T, cellTablets bool) {
+	ft := newFakeTopo(cellTablets)
 	fhc := newFakeHealthCheck()
 	t.Logf(`ft = FakeTopo(); fhc = FakeHealthCheck()`)
-	ctw := NewCellTabletsWatcher(topo.Server{Impl: ft}, fhc, "aa", 10*time.Minute, 5)
+	var ctw *TopologyWatcher
+	if cellTablets {
+		ctw = NewCellTabletsWatcher(topo.Server{Impl: ft}, fhc, "aa", 10*time.Minute, 5)
+	} else {
+		ctw = NewShardReplicationWatcher(topo.Server{Impl: ft}, fhc, "aa", "keyspace", "shard", 10*time.Minute, 5)
+	}
 	t.Logf(`ctw = CellTabletsWatcher(topo.Server{ft}, fhc, "aa", 10ms, 5)`)
```

Comment:
Please move the log line inside if-else, to show the called func name and params